src/components/screens/monitor/blockDetails/blockDetails.js

```diff
@@ -57,49 +57,49 @@ const getFields = (data, token, t, currentHeight) => ({
   },
   height: {
     label: t('Height'),
-    value: <CopyToClipboard value={data.height} />,
+    value: <CopyToClipboard value={data?.height} />,
   },
   date: {
     label: t('Date'),
     value: (
       <DateTimeFromTimestamp
-        time={data.timestamp * 1000}
+        time={data?.timestamp * 1000}
         token={tokenMap.BTC.key}
       />
     ),
   },
   confirmations: {
     label: t('Confirmations'),
-    value: currentHeight ? currentHeight - data.height : '-',
+    value: data?.height ? currentHeight - data.height : '-',
   },
   version: {
     label: t('Version'),
-    value: data.version,
+    value: data?.version,
   },
   generator: {
     label: t('Generated by'),
     value: (
       <Generator
-        generatorAddress={data.generatorAddress}
-        generatorUsername={data.generatorUsername}
+        generatorAddress={data?.generatorAddress}
+        generatorUsername={data?.generatorUsername}
```

Comment:
I think instead of having optional chaining on `data` in many places within the getFields function block, I think on line 42 this can be done so that even if data I undefined, it falls back to an empty object

`const getFields = (data = {}, token, t, currentHeight) => ({`